### PR TITLE
sweepbatcher: customize transaction labels

### PR DESCRIPTION
**Pull request by starius**
Previously sweepbatcher used `loop/labels.LoopOutBatchSweepSuccess` to assign transactions labels. The value is "BatchOutSweepSuccess -- $batch_id", which does not fit use cases outside of loop-out. Now there is option `WithTxLabeler` which sets a function used to generate the label.

#### Pull Request Checklist
- [ ] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

### CODE REVIEW

File: sweepbatcher/sweep_batch.go [hunk 1]
```diff
@@ -142,6 +141,10 @@ type batchConfig struct {
 	// external source of fee rates (FeeRateProvider).
 	noBumping bool
 
+	// txLabeler is a function generating transaction label. It is called
```
Comment (hieblmi) [inline sweepbatcher/sweep_batch.go:144]: nit: "generating a transaction label"

File: sweepbatcher/sweep_batcher.go [hunk 2]
```diff
@@ -258,6 +259,10 @@ type Batcher struct {
 	// ignored and fee bumping by sweepbatcher is disabled.
 	customFeeRate FeeRateProvider
 
+	// txLabeler is a function generating transaction label. It is called
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher.go:262]: nit: "generating a transaction label"

File: sweepbatcher/sweep_batcher.go [hunk 3]
```diff
@@ -272,6 +277,10 @@ type BatcherConfig struct {
...
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher.go:280]: nit: "generating a transaction label"

File: sweepbatcher/sweep_batcher.go [hunk 4]
```diff
@@ -291,6 +300,15 @@ func WithCustomFeeRate(customFeeRate FeeRateProvider) BatcherOption {
 	}
 }
 
+// WithTxLabeler sets a function generating transaction label. It is called
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher.go:303]: nit: "generating a transaction label"

File: sweepbatcher/sweep_batcher.go [hunk 5]
```diff
@@ -291,6 +300,15 @@ func WithCustomFeeRate(customFeeRate FeeRateProvider) BatcherOption {
...
+// before publishing a batch transaction. Batch ID is passed to the function.
+// By default, loop/labels.LoopOutBatchSweepSuccess is used.
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher.go:305]: nit: maybe move this comment to where the default value is set?

File: sweepbatcher/sweep_batcher_test.go [hunk 6]
```diff
@@ -399,6 +400,138 @@ func testFeeBumping(t *testing.T, store testStore,
 	}
 }
 
+// walletKitWrapper wraps a wallet kit and memorizes label of the most recent
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher_test.go:403]: nit: "the label"

File: sweepbatcher/sweep_batcher_test.go [hunk 7]
```diff
@@ -399,6 +400,138 @@ func testFeeBumping(t *testing.T, store testStore,
...
+	ctx, cancel := context.WithCancel(context.Background())
+
+	sweepStore, err := NewSweepFetcherFromSwapStore(store, lnd.ChainParams)
+	require.NoError(t, err)
+
+	walletKit := &walletKitWrapper{WalletKitClient: lnd.WalletKit}
+
+	batcher := NewBatcher(walletKit, lnd.ChainNotifier, lnd.Signer,
+		testMuSig2SignSweep, testVerifySchnorrSig, lnd.ChainParams,
+		batcherStore, sweepStore)
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher_test.go:435]: nit: newline here

File: sweepbatcher/sweep_batcher_test.go [hunk 8]
```diff
@@ -399,6 +400,138 @@ func testFeeBumping(t *testing.T, store testStore,
...
+	var runErr error
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher_test.go:436]: nit: var-block

File: sweepbatcher/sweep_batcher_test.go [hunk 9]
```diff
@@ -399,6 +400,138 @@ func testFeeBumping(t *testing.T, store testStore,
...
+
+	// Define dummy tx labeler, always returning "test".
+	txLabeler := func(batchID int32) string {
+		return "test"
+	}
+
+	// Now try it with option WithTxLabeler.
+	batcher = NewBatcher(walletKit, lnd.ChainNotifier, lnd.Signer,
+		testMuSig2SignSweep, testVerifySchnorrSig, lnd.ChainParams,
+		batcherStore, sweepStore, WithTxLabeler(txLabeler))
```
Comment (hieblmi) [inline sweepbatcher/sweep_batcher_test.go:512]: nit: newline

**Review by hieblmi — APPROVED** [review]
Thank for the improvement. LGTM, only minor nits.

**Review by bhandras — APPROVED** [review]
LGTM, pending comments from @hieblmi 🎉

File: sweepbatcher/sweep_batch.go (hunk 1 above)
Comment (starius) [reply sweepbatcher/sweep_batch.go]: Fixed. Thanks!

File: sweepbatcher/sweep_batcher.go (hunk 2 above)
Comment (starius) [reply sweepbatcher/sweep_batcher.go]: Fixed. Thanks!

File: sweepbatcher/sweep_batcher.go (hunk 3 above)
Comment (starius) [reply sweepbatcher/sweep_batcher.go]: Fixed. Thanks!

File: sweepbatcher/sweep_batcher.go (hunk 4 above)
Comment (starius) [reply sweepbatcher/sweep_batcher.go]: Fixed. Thanks!

File: sweepbatcher/sweep_batcher_test.go (hunk 6 above)
Comment (starius) [reply sweepbatcher/sweep_batcher_test.go]: Fixed

File: sweepbatcher/sweep_batcher_test.go (hunk 7 above)
Comment (starius) [reply sweepbatcher/sweep_batcher_test.go:438]: Fixed

File: sweepbatcher/sweep_batcher_test.go (hunk 8 above)
Comment (starius) [reply sweepbatcher/sweep_batcher_test.go]: Fixed

File: sweepbatcher/sweep_batcher_test.go (hunk 9 above)
Comment (starius) [reply sweepbatcher/sweep_batcher_test.go:519]: Fixed

**Pull request merged**